### PR TITLE
Remove memory leaks

### PR DESCRIPTION
**Pull request by emcrisostomo**
Removed some memory leaks and an unused variable

### CODE REVIEW

**Comment by yujitach** [comment]
Thanks as always. 
- I'm happy to merge the bug fix to MenuMeterPowerMate.m. 
- As for the fixes to MenuMeterDefaults.m, I think we should just rewirte it to NSUserDefaults, not CFPreferencesCopyValue... does the latter readlly needed? 
- As for the fixes to MenuMeterNetExtra.m, I think the issue is not about the unused variables but the bug #14. Could you fix #14 if you have time?

File: Common/MenuMeterPowerMate.m [hunk 1]
```diff
@@ -154,24 +154,27 @@ - (id)init {
 										   matchingDict,
 										   DeviceMatched, (__bridge void *)(self), &deviceMatchedIterator);
 	if (err != KERN_SUCCESS) {
+        CFRelease(terminatedDict);
```
Comment (romuald) [inline Common/MenuMeterPowerMate.m]: tab/space mixup here it seems

**Comment by emcrisostomo** [comment]
Hi @yujitach, I understand you're proposing improvements but they're out of scope. I sent the PR to fix a memory leak: other improvements can be done on other branches.

In the meantime I'll merge this into my master branch.

**Comment by yujitach** [comment]
I'm now confused by the original code in MenuMeterPowerMate.m.
The API document says that `IOServiceAddMatchingNotification` always consumes one retain count of `matchingDict`. It doesn't say that it doesn't consume the retain count if the function returns an error. Then, doesn't `CFRelease(matchingDict)` after `if (err != KERN_SUCCESS)` over-release things?

**Comment by emcrisostomo** [comment]
Hi @yujitach, I think you're right. I just checked the documentation as well. I'm going to add a change set to this PR.

**Comment by emcrisostomo** [comment]
Just added `7b7ec3c`

**Comment by luckman212** [comment]
How's this PR working out? Could you guys release a precompiled build to get some wider testing?

File: Common/MenuMeterPowerMate.m (hunk 1 above)
Comment (emcrisostomo) [reply Common/MenuMeterPowerMate.m]: not really, just aligned method arguments vertically.

**Comment by emcrisostomo** [comment]
Hi @luckman212,

As stated above, I think we shouldn't conflate more things into a single issue, so I won't be acting further on this PR. It's up to the repository maintainer to decide whether to merge it or not.

However, I'm maintaining a [fork] of this repository where this PR has been merged into, and there's a dmg image (with a pkg installer) attached to the [releases] I'll be shipping. Feel free to try it and give me feedback. You can even build it on your own system with a single command: `make`.

[fork]: https://github.com/emcrisostomo/MenuMeters
[releases]: https://github.com/emcrisostomo/MenuMeters/releases/latest

**Comment by luckman212** [comment]
Thanks @emcrisostomo, I've just compiled & installed your fork on 2 machines. Will give it a test.

**Comment by emcrisostomo** [comment]
Thanks @luckman212

**Comment by luckman212** [comment]
5 months later and I've had no issues at all with this PR.  Running on 10.13.4 now. Not sure about everyone else but would like to see this merged...

**Pull request merged**